### PR TITLE
Migrate AD cmdlets to use Autorest sdk and cmdlet and test fixes.

### PR DESCRIPTION
**Pull request by shuagarw**
- Use Graph.RBAC autorest sdk.
- Fix GetGroupMembers cmdlet paging issue.
- Minor tests and cmdlet fixes.

### CODE REVIEW

**Comment by markcowl** [comment]
@shuagarw Conflicts, please rebase.

**Comment by hovsepm** [comment]
on-demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1040/

**Comment by hovsepm** [comment]
@shuagarw please regenerate wxi file. You have introduced a new dependency (ClientRuntime.Azure) to couple of package projects

**Comment by shuagarw** [comment]
On-Demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1041/

**Comment by shuagarw** [comment]
On-Demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1042/

**Comment by markcowl** [comment]
signing job:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/548/

File: src/Common/Commands.Common/packages.config [hunk 1]
```diff
@@ -11,6 +11,7 @@
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.0" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
   <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure.TestFramework" version="1.3.4-preview" targetFramework="net45" />
```
Comment (markcowl) [inline src/Common/Commands.Common/packages.config]: This should not be included as a dependency in shipping code.  Please remove.

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/GetAzureADApplicationCommand.cs [hunk 2]
```diff
@@ -52,21 +52,23 @@ public override void ExecuteCmdlet()
...
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.DisplayName.StartsWith(DisplayNameStartWith));
                 }
                 else if (!string.IsNullOrEmpty(IdentifierUri))
                 {
-                    parameters.IdentifierUri = IdentifierUri;
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.IdentifierUris.Contains(IdentifierUri));
                 }
 
-                WriteObject(ActiveDirectoryClient.GetApplicationWithFilters(parameters), enumerateCollection: true);
+                WriteObject(ActiveDirectoryClient.GetApplicationWithFilters(odataQueryFilter), enumerateCollection: true);
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/GetAzureADApplicationCommand.cs]: @shuagarw  Is there a change in the output of thic cmdlet?  If so, we will need to document the change in release notes.

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/GetAzureADGroupMemberCommand.cs [hunk 3]
```diff
@@ -38,6 +40,12 @@ public override void ExecuteCmdlet()
...
 
+            PSADObject group = ActiveDirectoryClient.FilterGroups(options).FirstOrDefault();
+            if (group == null)
+            {
+                throw new KeyNotFoundException(string.Format(ProjectResources.GroupDoesntExist, GroupObjectId));
+            }
+
             do
             {
                 WriteObject(ActiveDirectoryClient.GetGroupMembers(options), true);
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/GetAzureADGroupMemberCommand.cs]: @shuagarw Same comment:  if there is a change here, we need to document it for the release notes

File: src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/PSADGroup.cs [hunk 4]
```diff
@@ -16,8 +16,6 @@ namespace Microsoft.Azure.Commands.Resources.Models.ActiveDirectory
 {
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/PSADGroup.cs]: @shuagarw We need to document this change in release notes

File: src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/PSADUser.cs [hunk 5]
```diff
@@ -17,7 +17,5 @@ namespace Microsoft.Azure.Commands.Resources.Models.ActiveDirectory
     public class PSADUser : PSADObject
     {
         public string UserPrincipalName { get; set; }
-
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/PSADUser.cs]: @shuagarw We need to document this change in release notes.

File: src/Common/Commands.Common/packages.config (hunk 1 above)
Comment (shuagarw) [inline src/Common/Commands.Common/packages.config]: Done.

**Comment by markcowl** [comment]
@shuagarw You will need to update this PR, there are merge conflicts

**Comment by shuagarw** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1051/

**Comment by shuagarw** [comment]
Another one: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1052/

**Comment by markcowl** [comment]
on demand run here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1055/

**Pull request merged**